src/pawthos/structs/data.rs

```diff
@@ -60,13 +61,16 @@ impl Data {
     /// # summary
     /// # }
     /// ```
-    pub async fn with_user_read<R, F>(&self, user_id: UserId, f: F) -> R
+    pub async fn with_user_read<R, F>(&self, user_id: UserId, f: F) -> Result<R, MimicError>
```

Comment:
This means if F returns a result, you return Result<Result<T, E>, MimicError> - if you require all Fs return a result who's error is MimicError, you can flatten it inside this function, rather than callers having to do it.